### DISCUSSION TITLE
Add Custom Device Support

### DISCUSSION THREAD

**Discussion by ghost** [Dynatrace/dynatrace-configuration-as-code#342] [Archive] [answered]
Is it possible to add custom device support to monaco? The issue I see is that it only has a `POST` endpoint, no `GET`.

```
curl -X POST "https://abc123.live.dynatrace.com/api/v1/entity/infrastructure/custom/customDeviceID" \
-H "accept: application/json; charset=utf-8" \
-H "Authorization: Api-Token ***" \
-H "Content-Type: application/json; charset=utf-8" \
-d "{\"displayName\":\"coffeeMachine\",\"group\":\"myCustomDeviceGroup\",\"ipAddresses\":[\"10.0.0.1\"],\"listenPorts\":[80],\"favicon\":\"https://www.freefavicon.com/freefavicons/food/cup-of-coffee-152-78475.png\",\"configUrl\":\"http://coffee-machine.dynatrace.internal.com/coffeemachine/manage\",\"type\":\"coffee machine\",\"properties\":{},\"tags\":[\"office-linz\"],\"hostNames\":[\"coffee-machine.dynatrace.internal.com\"]}"
```

**Comment by tatomir146** [comment, marked as answer]
Hi @Dynatrace-Adam-Gardner, let's use discussions for this. What would be the use case for this?

**Reply by ghost** [reply]
I think this is better handled with the new entity type framework and metrics (settings-objects API)

One usecase would be to build a master assets list in Dynatrace by creating entity types and custom devices.

For example, I create a custom entity type called `switches` and create `n` custom devices that are of type `entity:switch`. Then I can attach metrics to those devices.

So if monaco can support the settings 2.0 schema, I think we can close this.

https://github.com/dynatrace-oss/dynatrace-monitoring-as-code/issues/107

**Reply by tatomir146** [reply]
This will be our next topic after we are done with settings 2.0. Closing this discussion now.
